docs/Secure-Coding-Guide-for-Python/CWE-693/CWE-184/compliant01.py

```diff
@@ -1,62 +1,38 @@
 # SPDX-FileCopyrightText: OpenSSF project contributors
 # SPDX-License-Identifier: MIT
-""" Compliant Code Example """
+"""Compliant Code Example"""
+
 import re
-import unicodedata
 import sys
 
-sys.stdout.reconfigure(encoding="UTF-8")
-
-
-class TagFilter:
-    """Input validation for human language"""
+if sys.stdout.encoding.lower() != "utf-8":
+    sys.stdout.reconfigure(encoding="UTF-8")
 
-    def filter_string(self, input_string: str) -> str:
-        """Normalize and validate untrusted string
 
-        Parameters:
-            input_string(string): String to validate
-        """
-        # normalize
-        _str = unicodedata.normalize("NFKC", input_string)
+def filter_string(input_string: str):
+    """Normalize and validate untrusted string
 
-        # modify, keep only trusted human words
-        _filtered_str = "".join(re.findall(r"[/\w<>\s-]+", _str))
-        if len(_str) - len(_filtered_str) != 0:
-            raise ValueError("Invalid input string")
+    Parameters:
+        input_string(string): String to validate
+    """
+    # TODO Canonicalize (normalize) before Validating
 
-        # validate, only allow harmless tags
-        for tag in re.findall("<[^>]*>", _str):
-            if tag not in ["<b>", "<p>", "</p>"]:
-                raise ValueError("Invalid input tag")
-        return _str
+    # validate, only allow harmless tags
+    for tag in re.findall("<[^>]*>", input_string):
+        if tag not in ["<b>", "<p>", "</p>"]:
+            raise ValueError("Invalid input tag")
+    # TODO handle exception
 
 
 #####################
 # attempting to exploit above code example
 #####################
 names = [
-    "YES 毛泽东先生",
-    "YES María Quiñones Marqués",
-    "YES Борис Николаевич Ельцин",
-    "YES Björk Guðmundsdóttir",
-    "YES 0123456789",
-    "YES <b>",
-    "YES <p>foo</p>",
-    "YES underscore_",
-    "YES dash-",
-    "NOK semicolon;",
-    "NOK noprint " + "\uFDD0",
-    "NOK noprint " + "\uFDEF",
-    "NOK <script" + "\uFDEF" + ">",
-    "NOK <script生>",
-    "NOK and &",
+    "毛泽东先生",
+    "dash-",
+    "<script" + "\ufdef" + ">",
+    "<script生>",
```

Comment:
We can change this part so that the code exploiting section is the same in both code examples.
```suggestion
    "YES 毛泽东先生",
    "YES dash-",
    "NOK <script" + "\ufdef" + ">",
    "NOK <script生>",
```

Reply:
will do